src/lib/completion.ts

```diff
@@ -1,131 +1,216 @@
+			this.do_summaryBot
+		);
+		const model = new ChatOpenAI({
+			openAIApiKey: this.openai_api_key,
+			modelName: this.model_name,
+			temperature: 0.0,
+			stop: ['\nObservation']
 		});
-
-		console.log(qdrant_results);
-
-		// eslint-disable-next-line @typescript-eslint/no-explicit-any
-		return qdrant_results.map((result: any) => {
-			return new Document({
-				pageContent: result.payload.pageContent,
-				metadata: result.payload.metadata
-			});
+		const runnable = RunnableSequence.from([
+			{
+				input: (values: InputValues) => {
+					return values.input;
+				},
+				intermediate_steps: (values: InputValues) => {
+					return values.steps;
+				}
+			},
+			this.formatMessages,
+			model,
+			this.customOutputParser.bind(this)
+		]);
+		const executor = new AgentExecutor({
+			agent: runnable,
+			tools: tools
 		});
+		this.executor = executor;
 	}
-
-	private async get_vector_response(query: string): Promise<string[]> {
-		console.log('Retrieving vector response from qdrant...');
-
-		const vector_response = await this.qdrant_similarity_search(query, 2);
-
-		console.log(vector_response);
-
-		console.log('Vector response retreived');
-
-		// eslint-disable-next-line @typescript-eslint/no-explicit-any
-		return vector_response.map((document: Document<Record<string, any>>) => {
-			return document.pageContent;
-		});
-	}
-
-	private generate_history(history: ChatHistory[]): BaseMessage[] {
-		return history.map((message: { content: string; type: ChatHistoryType }) => {
-			if (message.type == ChatHistoryType.AI) {
-				return new AIMessage({ content: message.content });
-			} else {
-				return new HumanMessage({ content: message.content });
+	private formatMessages = async (values: InputValues) => {
+		if (this.history.length != 0) this.history.pop();
+		const history =
+			this.history
+				.map((input: { type: string; content: string }) => {
+					return input.type + ': ' + input.content;
+				})
+				.join('\n') + '\n';
+		const tools = getTools(
+			this.qdrantClient,
+			this.collection_name,
+			this.embeddings_model,
+			this.summaryBot,
+			this.do_summaryBot
+		); //to do seperate function for this
+		const intermediateSteps = values.intermediate_steps
+			? (values.intermediate_steps as Array<AgentStep>)
+			: [];
+		const agentScratchpad = formatLogToString(intermediateSteps);
+		let toolString = '';
+		let toolNames = '';
+		for (let i = 0; i < tools.length; i++) {
+			toolString += '\n' + tools[i].name + ': ' + tools[i].description;
+			toolNames += tools[i].name + ', ';
+		}
+		toolNames = toolNames.slice(0, -1);
+		toolNames = toolNames.slice(0, -1);
+		toolString += '\n\n';
+		const system = SYSTEM.replace('{tool_names}', toolNames).replace('{tools}', toolString);
+
+		const formatted = [
+			system,
+			HISTORY,
+			history,
+			SUFFIX.replace('{input}', values.input),
+			agentScratchpad
+		].join('');
+		if (this.verbose) {
+			console.log(colors.fg.green, 'Model prompt: ', colors.style.reset);
```

Comment:
I wan't us to also be logging a timestamp here as well.

Reply:
Do you mean console.log(current_time)?

Reply:
Yeah, just log the time along with the text